uaclient/cli.py

```diff
@@ -346,28 +346,48 @@ def _attach_with_token(
     return 0
 
 
-@assert_not_attached
-@assert_root
-def action_attach_premium(args, cfg):
+def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
+    """Detect cloud_type and request a contract token from identity info.
+
+    :param cfg: Parsed configuration dictionary
+
+    :raise NonPremiumImageError: When not on a premium image type. Exits 0.
+    :raise UserFacingError: On unexpected connectivity issues to contract
```

Comment:
Should this say `ContractAPIError` instead of `UserFacingError`?

Reply:
I'm not going to make this change because I want @blackboxsw to confirm.

Reply:
UserFacingError is incorrect, it should be UrlError for connetivity issues (in serviceclient.py) and ContractAPIError if error in talking to the backend. I've updated the docstring